### ISSUE TITLE
WLANs are not displayed when WLAN is on, but GPS is off

### ISSUE THREAD

**Issue by sjvudp** [mvglasow/satstat#136]
I noticed that the WLAN tab remains empty when GPS is not enabled (but WLAN is). I see no functional dependency.
So why is that so?

**Comment by sm4rk0** [comment]
Strange but true. I confirm with v3.2.

**Comment by dnussb** [comment]
FWIW: on Cyanogenmod11 (Kitkat 4.4.4) it's not the case, but on LineageOS13 (Marshmallow 6.0.1). Roms are running on two seperate Galaxy Nexus devices, both have 3.2 (git-da69945) installed. Since the former is SELinux "Permissive" and the latter "Enforcing" I played with changing it, but no luck.

**Comment by sjvudp** [comment]
I just started to learn how apps work, and I can't program Java, but could the problem be this?:
`registerPhoneStateListener()` is called when GPS services (`ACCESS_FINE_LOCATION`) are granted. That method also registers listening for `LISTEN_DATA_CONNECTION_STATE` (WLAN, I guess), also. Maybe the latter should be done in the `else` part (which uses mixed tab and spaces, BTW) also:
```
if (ContextCompat.checkSelfPermission(this, Manifest.permission.ACCESS_FINE_LOCATION) == PackageManager.PERMISSION_GRANTED)
        	registerPhoneStateListener();
        else
        	permsRequested[Const.PERM_REQUEST_PHONE_STATE_LISTENER] = true;

```
```
private void registerPhoneStateListener() {
		telephonyManager.listen(mPhoneStateListener, (LISTEN_CELL_INFO | LISTEN_CELL_LOCATION | LISTEN_DATA_CONNECTION_STATE | LISTEN_SIGNAL_STRENGTHS));
}
```
Maybe it's all about permissions...

**Comment by goldfndr** [comment]
One can see from [the upcoming Tasker Release Notes](http://tasker.dinglisch.net/changes/nextversion.html) that it'll display a warning if one is using Android 8.1 and trying to detect WiFi APs without Location Services being enabled.

I recall reading something a few months ago that Google was disallowing WiFi scans if Location Services was turned off, as a WiFi scan could be correlated with a database to detect location (by a perhaps malicious third-party app}, but I'm unable to find it again.

**Comment by sm4rk0** [comment]
It makes sense, but I can reproduce this on Lineage 13 (Android 6.0.1).

**Comment by dnussb** [comment]
I've just learned from what goldfndr points out, that there's been a design change since Android 6.x regarding LocationServices. The reasoning, as I realize, is something like this: "Since we are collecting so much data giving flesh to our giant geoip SSID database, we should ask users every time they connect wifi (and bluetooth, see URL below) if they really would like to reveal their location." So for my conclusion, Google became someway concerned about the Frankenstein monster they've just created… there's been something like a "_nomap" opt-out for SSIDs as a workaround in the first place, but this doesn't change the evil thing at all.

https://android.stackexchange.com/questions/160479/why-do-i-need-to-turn-on-location-services-to-pair-with-a-bluetooth-device

**Comment by goldfndr** [comment]
It's not just Google. There's also WiGLE.net, MozStumbler, and openBmap.org, to name a few off the top of my head.

**Comment by dnussb** [comment]
@goldfndr 
Thank you, I really didn't know and were unable to notice any creative advantage in such databases, hence my grim picture. It's all about having a choice, but since I'm technically poorly educated, I don't have much of it. Thanks again, your input eases my mind.

**Comment by mvglasow** [comment]
Correct, Android internally has no such thing as “GPS permission” but rather a location permission, which comes at two levels (coarse and fine). Since wifis can be used to determine the location of the device, accessing the list of wifis requires fine location permission (as the location is typically accurate to tens of meters). Listing cells, as far as I can remember, requires coarse location permission, as cell-based location is typically in the order of magnitude of hundreds to thousands of meters.

Does that answer your original question?
